jsoniter-scala-macros/src/main/scala/com/github/plokhotnyuk/jsoniter_scala/macros/JsonCodecMaker.scala

```diff
@@ -151,6 +156,7 @@ class CodecMakerConfig private (
       transientDefault = transientDefault,
       transientEmpty = transientEmpty,
       transientNone = transientNone,
+      requireCollectionFields = requireCollectionFields,
```

Comment:
I'm not sure if we should consider `Option` included as a collection. It'd be weird that at least something marked as `Option[Int]` would be require if `requireCollectionFields = true`.

Reply:
This flag doesn't work for for fields which are optional or have defined default values.